themes/default/content/blog/deploy-ai-ml-vercel-app/index.md

```diff
@@ -0,0 +1,204 @@
+
+# Configure vercel token as secret
+pulumi config set --secret token <vercel_token>
+
+```
+
+### 4. Deploy to Vercel
+
+Now, you're set up to deploy your Vue app to Vercel. 
+
+```bash
+# Finally, run `pulumi up` to deploy!
+pulumi up
+```
+
+### 6. Interact with your chatbot
+
+Once your web app is running, you can visit the website, send prompts and see the response in real-time!
+
+> Figure 1. A prompt and response screenshot from our Katwalk Frontend
+![chatbot-webapp](./chatbot-webapp.png)
+
+### 7. Tear down resources
+
+Now that you've tested out this code and you can see the chatbot in action, remember to deprovision your deployment by running: 
+
+```bash
+pulumi destroy
+```
+
+## Explaining the Pulumi code
+
+If you'd like to learn about what the Pulumi code is doing, then stick around. In this section, we'll dive into the Python Pulumi code which is responsible for deploying the Vue.js project to Vercel using Pulumi. Let's break down the code step by step to understand how it works. 
+
+### 1. Import the necessary modules
+
+```python
+import pulumi
+import pulumiverse_vercel as vercel
+```
+The code begins by importing the required modules. `pulumi` is the primary Pulumi Python library, while `pulumiverse_vercel` is a Pulumi provider for Vercel. These modules provide the necessary functionality to define and manage our web app infrastructure.
+
+### 2. Configuration settings
+
+```python
+config = pulumi.Config()
+token = config.require_secret("token")
+repoName = config.require("repoName")
+repoType = config.require("repoType")
+```
+Here, we retrieve configuration settings using Pulumi's configuration management. Configuration values like `token`, `repoName`, and `repoType` are essential for setting up the Vercel deployment. These were the values that you provided using the `pulumi config set` command before deploying the stack.
+
+### 3. Vercel provider
+
+```python
+provider = vercel.Provider("vercel-provider",
+    api_token = token
+)
+```
+This part initializes the Vercel provider with the Vercel API token. The provider is used to authenticate and interact with the Vercel platform programmatically.
+
+### 4. Creating a Vercel project
+
+```python
+project = vercel.Project("vercel-project", 
+    name = "vercel-git-project",
+    framework = "vue",
+    git_repository = vercel.ProjectGitRepositoryArgs(
+        repo = repoName,
+        type = repoType
+    ),
+    root_directory = "./src/app/katwalk-frontend",
+    opts = pulumi.ResourceOptions(
+        provider = provider
+    )
+)
+```
+Here, we define a Vercel project named `vercel-git-project`. The `git_repository` parameter specifies the Git repository details, including `repoName` and `repoType`. We also associate this project with the Vercel provider using `opts`.
+
+### 5. Setting Environment Variables
+
+```python
+environment = vercel.ProjectEnvironmentVariable("vercel-env",
+    project_id = project.id,
+    key = "VUE_APP_BACKEND_DNS",
```

Comment:
you should provide more instructions about how to get this backend DNS. 

this should be coming from the output of the katwalk backend output. 

in fact, what you should have done is have this value be passed in as a pulumi config. 

that way you can do a https://www.pulumi.com/learn/building-with-pulumi/stack-references/